plugins/quota/actor.go

```diff
@@ -0,0 +1,95 @@
+package quota
+
+import (
+	"os"
+	"path"
+
+	"github.com/gluster/glusterd2/glusterd2/daemon"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+	"github.com/gluster/glusterd2/glusterd2/xlator"
+	"github.com/gluster/glusterd2/pkg/errors"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type quotadActor struct{}
+
+const (
+	quotaDaemonKey = "enable"
+)
+
+// isQuotadAffectedNode returns true if there are local bricks of volume on
+// which quota is enabled
+func isQuotadAffectedNode(volumes []*volume.Volinfo) bool {
+	for _, v := range volumes {
+		if !isQuotaEnabled(v) {
+			continue
+		} else if v.State != volume.VolStarted {
+			continue
+		} else {
+			bricks := v.GetLocalBricks()
+			if len(bricks) > 0 {
+				return true
+			}
+			continue
+		}
+	}
+	return false
+}
+
+func (actor *quotadActor) Do(v *volume.Volinfo, key, value string, logger log.FieldLogger) error {
+	if key != quotaDaemonKey {
+		return nil
+	}
+	quotadDaemon, err := NewQuotad()
+	if err != nil {
+		return err
+	}
+	// Create pidfile dir if not exists
+	if err := os.MkdirAll(path.Dir(quotadDaemon.pidfilepath),
+		os.ModeDir|os.ModePerm); err != nil {
+		return err
+	}
+
+	// Create logFiledir dir
+	if err := os.MkdirAll(path.Dir(quotadDaemon.logfilepath),
+		os.ModeDir|os.ModePerm); err != nil {
+		return err
+	}
+
+	volumes, err := volume.GetVolumes()
+	if err != nil {
+		logger.WithError(err).Error("failed to get volumes")
+		return err
+	}
+	// Check for making changes to quotad on this node
+	if !isQuotadAffectedNode(volumes) {
+		// This condition is for disable
```

Comment:
I'm not sure how you figure out that this is disable without checking the user input - key and value variables

Reply:
For both disable and enable we need to do the same (restart). 

Lets take a case where we have only one volume on a node. it goes to the else part were we stop and the start. stop fails with pidfile not found and start , starts the daemon. 
if we issue stop on this volume, then it goes to the if part where we stop it. (Nit: need not correct the log message)

If we have two volumes in the same node. then they share the quotad (difference lies only in the volfile) during start for the second volume, we go to the else part and we stop the quotad and then we go to the daemon start and start the quotad.(now it will start with the volfile changes. ie its common for both volumes) 
During the stop for the second volume, we again go to the else part were we stop and start with the volfile changes for a single volume.

Reply:
I'll get some more confidence is there were a e2e testcase for it. Can you please add one ? A simple test which enables quota, checks if quotad process is running, disables quota, checks if quoatd process is running.

Reply:
I had a error string missing for quotad disable on disabled volume. have added that part. I have updated the e2e testcase for quota enable with what you have asked for.